### ISSUE TITLE
Loaded Image for Button is Too Padded

### ISSUE THREAD

**Issue by yishayw** [apache/royale-asjs#796]
```
<mx:Application xmlns:fx="http://ns.adobe.com/mxml/2009" xmlns:mx="library://ns.apache.org/royale/mx" 
	applicationComplete="init()"
	width="600" height="400">
	<fx:Script>	
	<![CDATA[
		import mx.containers.HBox;
		import mx.controls.Button;

		private function init():void
		{
			var buttonContainer:HBox = new HBox();
			this.addChild(buttonContainer);
			var minimizeButton:Button = new Button();
			minimizeButton.width = minimizeButton.height = 16;
			minimizeButton.icon = "icon_close_up.png"
			buttonContainer.addChild(minimizeButton);
			var maximizeButton:Button = new Button();
			maximizeButton.width = maximizeButton.height = 16;
			maximizeButton.icon = "icon_close_up.png";
			buttonContainer.addChild(maximizeButton);
		}

	]]>
	</fx:Script>
</mx:Application>
```

Above application has an overflow on the HBox after the icons are loaded. This seems to be caused by these lines in Button.as
```
var img:HTMLImageElement = event.target as HTMLImageElement;
		element.style["padding-left"] = String(img.naturalWidth+4)+"px";
```

I'm not sure what the purpose of this was (added by @pentapache ).

**Comment by aharui** [comment]
I suspect that code is to put 4 pixels of padding between the image and the label.

I think you may just be running into another deferred layout situation like you did in #790.  The Button will be measured before the image gets loaded so will need to be re-measured when the image loads.

**Comment by yishayw** [comment]
The thing is the style.width and style.height are correct, so I don't think measuring will help. What's overflowing is the button contents (bg image in this case).

**Comment by yishayw** [comment]
I just confirmed call BoxLayout.measure() doesn't change anything because explicitWidth is set on the children.

**Comment by nihavend** [comment]
I have a similar issue related with icon image of button. Button size iş dider than the image size, behaves diffrent compared with flex.

**Comment by yishayw** [comment]
> I have a similar issue related with icon image of button. Button size iş dider than the image size, behaves diffrent compared with flex.

Actually, here it seems to be the other way around. The button size (width: 16px) is more narrow than the size calculated by Chrome (21px). I was reading that `box-sizing: border-box; `should prevent that but that doesn't seem to be the case. I guess this is a css issue we'll need to explore.

**Comment by aharui** [comment]
Ah, sorry, I didn't notice that the button had explicit width/height.  Is `element` an <img> or <span> or <button>? Possibly the fix is to detect that there is no label and set padding to 0.  The 4 is probably a guess at standard gap between elements when you have <img><span>

**Comment by yishayw** [comment]
Closed with  9d6252d

**Comment by nihavend** [comment]
Icon on the button fits good now but i think some styling is still needed for buttons with icon image.

flex : 
![image](https://user-images.githubusercontent.com/5983818/78425125-2c080700-767b-11ea-983d-005c06a20270.png)

royale :

![image](https://user-images.githubusercontent.com/5983818/78427756-c0bf3480-767c-11ea-922e-26838d428054.png)
